it/test/uk/gov/hmrc/alcoholdutycalculator/rates/RatesIntegration_2023_1Spec.scala

```diff
@@ -0,0 +1,283 @@
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, rate)
+          }
+        }
+      }
+
+      "For band F" - {
+        val rate: Option[Double]   = Some(31.64d)
+        val taxTypes: Seq[String]  = Seq("341", "343", "344", "345")
+        val rateCount: Int         = 4
+        val rangeDetailsCount: Int = 4
+        val rangesCount: Int       = 4
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, rate)
+          }
+        }
+      }
+
+      "For band G" - {
+        val rate: Option[Double]   = Some(8.42d)
+        val taxTypes: Seq[String]  = Seq("351", "352", "353", "354", "355")
+        val rateCount: Int         = 5
+        val rangeDetailsCount: Int = 5
+        val rangesCount: Int       = 5
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, rate)
+          }
+        }
+      }
+
+      "For band H" - {
+        val rate: Option[Double]   = Some(19.08d)
+        val taxTypes: Seq[String]  = Seq("356")
+        val rateCount: Int         = 1
+        val rangeDetailsCount: Int = 1
+        val rangesCount: Int       = 1
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, rate)
+          }
+        }
+      }
+
+      "For band I" - {
+        val rate: Option[Double]   = Some(8.78d)
+        val taxTypes: Seq[String]  = Seq("357")
+        val rateCount: Int         = 1
+        val rangeDetailsCount: Int = 1
+        val rangesCount: Int       = 2
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, rate)
+          }
+        }
+      }
+
+      "For band J" - {
+        val rate: Option[Double]   = Some(19.08d)
+        val taxTypes: Seq[String]  = Seq("358", "359", "360")
+        val rateCount: Int         = 3
+        val rangeDetailsCount: Int = 4
+        val rangesCount: Int       = 4
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, rate)
+          }
+        }
+      }
+
+      "For band K" - {
+        val taxTypes: Seq[String]  = Seq("371", "372", "373", "374", "375", "376", "377", "378", "379", "380")
+        val rateCount: Int         = 10
+        val rangeDetailsCount: Int = 11
+        val rangesCount: Int       = 12
+
+        "service rates endpoint must" - {
+          "show the correct rate" in {
+            verify(rateBandList, taxTypes, rateCount, rangeDetailsCount, rangesCount, None)
+          }
+        }
+      }
+    }
+  }
+
```

Comment:
Might be worth testing the total number of rateCount, rangeDetailsCount and rangesCount to make sure there is nothing extra in the file